### PR TITLE
[lang/c++] Make new LLVM version compile with unused variable

### PR DESCRIPTION
**Pull request by gaschler**
LLVM is making the unused variable warning a default: https://github.com/llvm/llvm-project/commit/92023b15099012a657da07ebf49dd7d94a260f84 This will make grpc compile after this LLVM update.

(release notes: no)

### CODE REVIEW

**Comment by gaschler** [comment]
@drfloob please take a look (feel free to re-assign)

File: src/core/lib/channel/connected_channel.cc [hunk 1]
```diff
@@ -548,7 +548,7 @@ ArenaPromise<ServerMetadataHandle> MakeClientCallPromise(
         }
         if (!done_recv_messages) {
           auto p = recv_messages();
-          if (auto* r = p.value_if_ready()) {
+          if ([[maybe_unused]] auto* r = p.value_if_ready()) {
```
Comment (yijiem) [inline src/core/lib/channel/connected_channel.cc:551]: `[[maybe_unused]]` is a C++17 attribute. I don't think we have moved to C++17 yet. And if initialization is C++17 extension too, see: https://godbolt.org/z/Mc5TTMe8T. So I suggest changing this code to:
```
auto* r = p.value_if_ready();
if (r) {
...
} else {
...
}
```

Cc @ctiller

Comment (yashykt) [reply src/core/lib/channel/connected_channel.cc:551]: probably just replace this with `p.value_if_ready() != nullptr`

Comment (gaschler) [reply src/core/lib/channel/connected_channel.cc]: Done (first option), please take another look.

**Review by yijiem — APPROVED** [review]

File: src/core/lib/channel/connected_channel.cc (hunk 1 above)
Comment (yijiem) [reply src/core/lib/channel/connected_channel.cc]: Ah, sorry, I think @yashykt's suggestion is fair, i.e. change this to `if (p.value_if_ready() != nullptr) {`. At least we should not depend on implicit bool conversion.

Comment (gaschler) [reply src/core/lib/channel/connected_channel.cc]: Done, please take another look

Comment (ctiller) [reply src/core/lib/channel/connected_channel.cc]: This could just be spelled `if (p.ready()) {`

Comment (gaschler) [reply src/core/lib/channel/connected_channel.cc]: Done, I now went with p.ready(), please take another look.
(Also, if you can trigger the remaining checks, that would be helpful, I'm unfamilar with how PRs are tested and merged in the grpc repo.)

**Review by veblush — APPROVED** [review]

**Pull request merged**